disbot.js

```diff
@@ -0,0 +1,323 @@
+        voice_channels.push(channel);
+      }
+    }
+
+    // for each @mention, move them around X times
+    for(var mem_index in members){
+      var member = members[mem_index];
+    }
+  }
+
+/*** end command functions ***/
+
+/*** ready status announcement ***/
+  client.on('ready', () => {
+    console.log('Ready!\n');
+  });
+/*** end ready status announcement ***/
+
+/**** listeners ****/
+  /*** direct response listener ***/
+    client.on('message', message => {
+      // ignore self-posts
+      if (message.author.bot){return;}
+
+      // useful attributes
+      var username = message.author.username;
+      var text_channel = message.channel;
+      var voice_channel = message.member.voiceChannel;
+      var full_message = message.content;
+      var cmd = '';
+      var args = [];
+      var attachments = message.attachments; // type = Discord.js[Collection]
+
+      console.log(attachments);
+      // var props = Object.keys(attachments);
+      // console.log('trying to print props');
+      // try {
+      //   for(var prop in props){
+      //     console.log(prop);
+      //   }
+      // } catch(err) {
+      //   console.log(err);
+      // }
+      console.log('now processing command');
+
+      // get arguments & set cmd // cleaned up message.content
+      var pre_args = full_message.trim().split(/ +/g);
+      try {
+        cmd = pre_args[0].trim().toLowerCase();
+      } catch(err) {
+        console.log(err);
+      }
+      try{
+        args = pre_args.slice(1,pre_args.length);
+      } catch(err) {
+        console.log(err);
+      } finally{
+        console.log('args: ' + args.toString());
+      }
+
+
+      if(isReady){
+        isReady = false;
+
+        switch(cmd){
+          /* voice commands */
+          case '!horn':
+            isReady = false;
+            var voiceChannel = message.member.voice.channel;
+            if(voiceChannel){
+              voiceChannel.join().then(connection =>{
+                const dispatcher = connection.play(audio_map[cmd]);
+                dispatcher.on("end", end => {
+                  message.member.voice.channel.leave();
+                });
+              }).catch(err => console.log(err));
+            } else {
+              message.reply('You need to join a voice channel first!');
+            }
+            break;
+          /* text commands */
+          case '!hello':
+            text_channel.send(text_map[cmd]);
+            break;
+          /* direct commands */
+          case '!roll':
+            var maybe_roll_max = parseInt(args.pop());
+            var roll_max = isNaN(maybe_roll_max) ? 100 : maybe_roll_max;
+            roll(message, roll_max);
+            break;
+          case '!yt':
+            var link = args.shift();
+            yt(voice_channel, link, username, message);
+            break;
+          /* fall-through */
+          case '/join':
+          case '!stop':
+            message.reply('Stopping boss!');
+            break;
```

Comment:
```suggestion
```

I'd handle the `!stop` command in just one location, and move `message.reply('Stopping boss!');` to the other handler for this special command